src/main/java/bleuauction/bleuauction_be/server/config/SecurityConfig.java

```diff
@@ -1,47 +1,88 @@
 import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.annotation.web.configurers.HeadersConfigurer;
 import org.springframework.security.crypto.factory.PasswordEncoderFactories;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.web.cors.CorsConfigurationSource;
-import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
 
 
 @Slf4j
 @RequiredArgsConstructor
 @EnableWebSecurity
+@EnableMethodSecurity(prePostEnabled = true)
 @Configuration
 public class SecurityConfig {
+  private final JwtUtils jwtUtils;
+  private final APIUserDetailsService apiUserDetailsService;
   private final CorsConfigurationSource corsConfigurationSource;
 
   @Bean
-  public SecurityFilterChain filterChain(HttpSecurity http,
-                                         HandlerMappingIntrospector introspector) throws Exception {
+  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+
+    AuthenticationManagerBuilder authenticationManagerBuilder = http.getSharedObject(AuthenticationManagerBuilder.class);
+    authenticationManagerBuilder
+            .userDetailsService(apiUserDetailsService)
+            .passwordEncoder(passwordEncoder());
+
+    AuthenticationManager authenticationManager = authenticationManagerBuilder.build();
+    http.authenticationManager(authenticationManager);
+
+    // 로그인 필터
+    APILoginFilter apiLoginFilter = new APILoginFilter("/generateToken");
+    apiLoginFilter.setAuthenticationManager(authenticationManager);
+
+    // 로그인 성공시 호출 핸들러 설정
+    APILoginSuccessHandler successHandler = new APILoginSuccessHandler(jwtUtils);
+    apiLoginFilter.setAuthenticationSuccessHandler(successHandler);
+
+    // UsernamePasswordAuthenticationFilter 앞쪽으로 APILoginFilter지정
+    http.addFilterBefore(apiLoginFilter, UsernamePasswordAuthenticationFilter.class);
+
+    // Token검증 필터 추가
+    http.addFilterBefore(
+            new TokenCheckFilter(apiUserDetailsService, jwtUtils),
+            UsernamePasswordAuthenticationFilter.class
+    );
+
+    // refreshToken 필터 등록 - JWT관련 다른 필터들 이전에 동작하도록 TokenCheckFilter 앞에 배치
+    http.addFilterBefore(new RefreshTokenFilter("/refreshToken", jwtUtils), TokenCheckFilter.class);
+
```

Comment:
이거 혹시 참고한 코드가 있을까요? 실제 필터를 추적하면서 따라가 보셨을까요?


successHandler의 경우에는 생성자에 있어도 괜찮지 않을까 하는 생각이 드네요

제가 생각했을떄는 이렇게로도 끝낼 수 있지 않을까 생각이 듭니다.
HttpSecurity 객체에서 getSharedObject로 객체를 받아오려는 이유도 궁금하네요.
😃
```suggestion
    http.authorizeHttpRequests(
                    authorizationManagerRequestMatcherRegistry ->
                            authorizationManagerRequestMatcherRegistry
                                    .anyRequest()
                                    .permitAll()
            ).logout(AbstractHttpConfigurer::disable)
            .csrf(AbstractHttpConfigurer::disable)
            .userDetailsService(apiUserDetailsService)
            .addFilterBefore(new APILoginFilter("/generateToken"), UsernamePasswordAuthenticationFilter.class)
            .addFilterBefore(new TokenCheckFilter(apiUserDetailsService, jwtUtils), UsernamePasswordAuthenticationFilter.class)

```

Reply:
Q. 이거 혹시 참고한 코드가 있을까요? 실제 필터를 추적하면서 따라가 보셨을까요?
A> 책보고 따라하면서 커스터 마이징 해았고 PostMan으로 log 찍히는거 보면서 정상적으로 작동하는지 확인은 해봤습니다. 

Q. successHandler의 경우에는 생성자에 있어도 괜찮지 않을까 하는 생각이 드네요
A>  그런생각을 못했었습니다. 저도 말씀하신게 훨씬 효율적인 거 같아서 로컬에서 APILoginFilter 생성자로 받아서 주입하는 식으로 고쳤습니다! 

Q. getSharedObject로 객체를 받아오려는 이유도 궁금하네요.
A> 그렇게 되는지 몰랐습니다ㅠ